### PR TITLE
Add `Repeat(Trail)` to operator list.

### PR DESCRIPTION
**Pull request by sherfert**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-cypher-578.surge.sh

File: modules/ROOT/pages/execution-plans/operators.adoc [hunk 1]
```diff
@@ -2772,6 +2772,58 @@ Batch size 128
 Total database accesses: 1, total allocated memory: 200
 ----
 
+[[query-plan-repeat]]
+== Repeat (Trail)
+// Repeat(Trail)
+
+Given a start node, the `Repeat(Trail)` operator will traverse quantified path patterns that cannot be solved with the xref::execution-plans/operators.adoc#query-plan-varlength-expand-all[`VarLengthExpand(All)`] operator
```
Comment (arnefischereit) [inline modules/ROOT/pages/execution-plans/operators.adoc]: I think this does not really talk about how it works. Maybe we can add the general behaviour wrt the plan structure:

```suggestion
Given a start node, the `Repeat(Trail)` operator will traverse quantified path patterns that cannot be solved with the xref::execution-plans/operators.adoc#query-plan-varlength-expand-all[`VarLengthExpand(All)`] operator.
Similar to an `Apply` operator, it takes a single row from the left-hand side and applies the operators on the right-hand side.
In contrast to `Apply`, however, it iteratively applies these operators in accordance with the quantifiers on the quantified path pattern.

In the following example, the operator will iterate twice and produce rows for both iterations.
```

File: modules/ROOT/pages/execution-plans/operators.adoc [hunk 2]
```diff
@@ -2772,6 +2772,58 @@ Batch size 128
...
+| |                +----+-----------------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
+| +NullifyMetadata |  7 |                                                     |             15 |   48 |       0 |                |                        |           |                     |
+| |                +----+-----------------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
+| +Repeat(Trail)   |  2 | (me) (...){1, 2} (friend)                           |             15 |   48 |       0 |          16784 |                    0/0 |     1.146 | Fused in Pipeline 2 |
+| |\               +----+-----------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
+| | +Filter        |  3 | not a.name = b.name AND isRepeatTrailUnique(anon_2) |              5 |   48 |     216 |                |                        |           |                     |
```
Comment (arnefischereit) [inline modules/ROOT/pages/execution-plans/operators.adoc:2865]: Should we also explain the cardinalities displayed for `Repeat` or is that still too much in flux / too internal?

**Review by arnefischereit — CHANGES_REQUESTED** [review]
I think we could describe the behaviour of `Repeat` a bit more. Have a look at what I suggested.

File: modules/ROOT/pages/execution-plans/operators.adoc (hunk 2 above)
Comment (sherfert) [reply modules/ROOT/pages/execution-plans/operators.adoc:2865]: We don't do that for any other operator, so I would not start doing that for the most complex operator we have :laughing:

File: modules/ROOT/pages/execution-plans/operators.adoc (hunk 1 above)
Comment (sherfert) [reply modules/ROOT/pages/execution-plans/operators.adoc]: I took your text, but replaced `iterate*` with `repeat*`, so it aligns with the operator name.
Let me know what you think.

Comment (arnefischereit) [reply modules/ROOT/pages/execution-plans/operators.adoc]: Makes sense to me :+1:

**Review by arnefischereit — APPROVED** [review]
I think, this is good now. :+1:

File: modules/ROOT/pages/execution-plans/operators.adoc [hunk 3]
```diff
@@ -2575,7 +2575,7 @@ Total database accesses: 215, total allocated memory: 3440
 == VarLength Expand All
 // VarLengthExpand(All)
 
-Given a start node, the `VarLengthExpand(All)` operator will traverse variable-length relationships.
+Given a start node, the `VarLengthExpand(All)` operator will traverse variable-length/quantified relationships.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/execution-plans/operators.adoc]: The same suggestion applies for the below examples of "variable-length/quantified".
```suggestion
Given a start node, the `VarLengthExpand(All)` operator will traverse variable-length and quantified relationships.
```

File: modules/ROOT/pages/execution-plans/operators.adoc [hunk 4]
```diff
@@ -2772,6 +2772,61 @@ Batch size 128
 Total database accesses: 1, total allocated memory: 200
 ----
 
+[[query-plan-repeat]]
+== Repeat (Trail)
+// Repeat(Trail)
+
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/execution-plans/operators.adoc:2826]: Add below title: 
```suggestion
_This feature was introduced in Neo4j 5.9._
```

**Review by JPryce-Aklundh — COMMENTED** [review]
Looks good.
I have a few suggestions (replacing "/" with "and", and adding a note when operator was introduced).
Also, we need to add this operator to:
- additions page: e.g., https://neo4j.com/docs/cypher-manual/current/deprecations-additions-removals-compatibility/#_new_features)
- operator summary page: https://neo4j.com/docs/cypher-manual/current/execution-plans/operator-summary/

Would it be possible for you to add to those pages in this PR?

**Comment by sherfert** [comment]
@JPryce-Aklundh I fixed all your comments.

I think we want to wait until https://github.com/neo4j/docs-cypher/pull/489 is merged, so that I can add links whenever I mention "quantified path patterns", WDYT?

**Review by JPryce-Aklundh — APPROVED** [review]
Thanks, @sherfert! Yes, sounds good 👍

**Comment by JPryce-Aklundh** [comment]
@sherfert - https://github.com/neo4j/docs-cypher/pull/489 has been merged :)

**Pull request merged**